### PR TITLE
For #6 Support ES6 export declarations

### PR DESCRIPTION
**Pull request by morlay**
#6 

```` js
import React from 'react';
export default React.createClass(...);
// or
export var Component = React.createClass(...);
// etc
````` 
This way is done.

Should we support this?
```` js
import React, { createClass } from 'react';
export default createClass(...);
````` 

And not sure, is it ok, to support ES6 module and CommonJS in one handler.

### CODE REVIEW

**Comment by fkling** [comment]
Sweet! I will take a close look later today. Thank you for this!

File: src/resolver/findExportedReactCreateClassCall.js [hunk 1]
```diff
@@ -47,6 +48,17 @@ function findExportedReactCreateClass(
     visitDoWhileStatement: ignore,
     visitForStatement: ignore,
     visitForInStatement: ignore,
+    visitExportDeclaration: function(path) {
+      path = resolveToCallExpression(path);
```
Comment (fkling) [inline src/resolver/findExportedReactCreateClassCall.js]: Mmh, do we have to introduce `resolveToCallExpression`? My approach would have been to just extract the path from `ExportDeclaration` that we are interested in, and run `resolveToValue` on that.

**Comment by fkling** [comment]
It doesn't seem like this would support support

```js
export {
  comeponent,
  something else
};
```

or does it?

> Should we support this?
>
    import React, { createClass } from 'react';
    export default createClass(...);

Not sure. I think if the need arises we can still that.

> And not sure, is it ok, to support ES6 module and CommonJS in one handler.

Seems fine to me.

File: src/resolver/findExportedReactCreateClassCall.js (hunk 1 above)
Comment (morlay) [inline src/resolver/findExportedReactCreateClassCall.js]: Sure, seems we can change here with 
``` js
path = resolveToValue(path.get('declaration'));
```
and add the conditions to `resolveToValue`
``` js
if (types.VariableDeclaration.check(node)) {
    return resolveToValue(path.get('declarations', 0))
  } else if (types.VariableDeclarator.check(node)) {
     if (node.init) {
       return resolveToValue(path.get('init'));
     }
  } 
```

I will push another commit to update the change. 

however, I feel the `resolveToValue` need to do some refact now, when we add a lot of conditions.

**Comment by morlay** [comment]
Cool. I feel we are not necessary to write like that.

With react-docgen's help, we can get docs so easy, in my last projects. 

So I put the docs tool in my [testing repo](https://github.com/morlay/react-semantic-ui) ([preview](http://morlay.github.io/react-semantic-ui/components-doc))

Recently, we need to update to ES6, that why I try to make the update for react-docgen.

**Comment by fkling** [comment]
Sorry for the late response! I included your changes in 55c1511cdd192f51f7183ed7ddc604ba7e86a68e and 9e8670c7b8197d7be98af8bf1d2b21d2763fced3 (I squashed the third one).

I also built on top of your changes and added support for the remaining `import` and `export` variations.

All these changes are release with v1.3.0.

Thanks for your help!

**Pull request closed without merging**